torchvision/models/detection/fcos.py

```diff
@@ -567,7 +567,7 @@ def forward(
         if self.training:
 
             if targets is None:
-                torch._assert(False, "targets should not be none when in training mode")
+                torch._assert(True, "targets should not be none when in training mode")
```

Comment:
I think you want this to be 
```suggestion
                torch._assert(False, "targets should not be none when in training mode")
```

Effectively you want ti to fail directly. If you pass True then the error message will never trigger.